src/System.Windows.Forms/src/System/Windows/Forms/Dialogs/CommonDialogs/FolderBrowserDialog.cs

```diff
@@ -373,13 +398,36 @@ private void SetOption(FILEOPENDIALOGOPTIONS option, bool value)
 
     private unsafe void GetResult(IFileOpenDialog* dialog)
     {
-        using ComScope<IShellItem> item = new(null);
-        dialog->GetResult(item);
-        if (!item.IsNull)
+        if (Multiselect)
+        {
+            using ComScope<IShellItemArray> itemArray = new(null);
+            dialog->GetResults(itemArray);
+            itemArray.Value->GetCount(out uint itemCount);
+            List<string> tempSelectedPaths = new();
+            for (uint itemIndex = 0; itemIndex < itemCount; itemIndex++)
+            {
+                using ComScope<IShellItem> item = new(null);
+                itemArray.Value->GetItemAt(itemIndex, item);
+                if (!item.IsNull)
+                {
+                    item.Value->GetDisplayName(SIGDN.SIGDN_FILESYSPATH, out PWSTR ppszName);
+                    tempSelectedPaths.Add(new(ppszName));
+                    Marshal.FreeCoTaskMem((nint)(void*)ppszName);
```

Comment:
This can be simplified a bit with a helper we have:
```suggestion
                    tempSelectedPaths.Add(ppszName.ToStringAndCoTaskMemFree());
```